src/Microsoft.ApplicationInsights/Metric.cs

```diff
@@ -447,7 +447,21 @@ public bool TryGetDataSeries(out MetricSeries series, bool createIfNotExists, pa
 
             for (int d = 0; d < dimensionValues.Length; d++)
             {
-                Util.ValidateNotNullOrWhitespace(dimensionValues[d], Invariant($"{nameof(dimensionValues)}[{d}]"));
+                var value = dimensionValues[d];
+                if (value == null)
+                {
+                    throw new ArgumentNullException(Invariant($"{nameof(dimensionValues)}[{d}]"));
```

Comment:
this is doing some weird string allocation. i think `nameof(dimensionValues)` should be a const